python-sdk/src/astro/files/locations/sftp/sftp.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import os
+from urllib.parse import urlparse
+
+from airflow.providers.sftp.hooks.sftp import SFTPHook
+
+from astro.constants import FileLocation
+from astro.exceptions import PermissionNotSetError
+from astro.files.locations.base import BaseFileLocation
+
+
+class SFTPLocation(BaseFileLocation):
+    """Handler SFTP object store operations"""
+
+    location_type = FileLocation.SFTP
+
+    @property
+    def hook(self) -> SFTPHook:
+        return SFTPHook(ssh_conn_id=self.conn_id) if self.conn_id else SFTPHook()
+
+    @property
+    def transport_params(self) -> dict:
+        """Get SFTP credentials in order to access remote file system"""
+        client = self.hook.get_connection(self.conn_id)
+        extra_options = client.extra_dejson
+        if "key_file" in extra_options:
+            key_file = extra_options.get("key_file")
+            return {"connect_kwargs": {"key_filename": key_file}}
+        elif client.password:
+            return {"connect_kwargs": {"password": client.password}}
+        raise PermissionNotSetError("SFTP credentials were not set in the connection.")
+
+    @property
+    def paths(self) -> list[str]:
```

Comment:
@rajaths010494 I see a lot of code replication in this PR and other PR of FTP. Can refactor on or the other?

Reply:
+1 for the file path suggestion

Reply:
will change the path

Reply:
+1

Reply:
changed